synapse/handlers/register.py

```diff
@@ -149,7 +162,7 @@ def register(
                     user_id = None
                     token = None
                     attempts += 1
-                    if attempts > 5:
+                    if attempts > 50:
```

Comment:
In that case, I'm not sure I'm entirely happy with the "try it and see if we get a SynapseError" approach to this. Is there really nothing else which could cause a SynapseError in that code block? I guess I'd prefer an approach which checked more directly for duplicates -- in which case I think you could remove the limit altogether.